pkg/cmd/repo/rename/rename.go

```diff
@@ -104,29 +108,72 @@ func renameRun(opts *RenameOptions) error {
 			"Rename %s to:", ghrepo.FullName(currRepo)), ""); err != nil {
 			return err
 		}
+
+		if !opts.RenameLocalDir {
+			var confirmed bool
+			if confirmed, err = opts.Prompter.Confirm("Would you also like to rename the local directory to the same name?", false); err != nil {
+				return err
+			}
+			if confirmed {
+				opts.RenameLocalDir = true
+			}
+		}
 	}
 
 	if opts.DoConfirm {
+		confirmMessage := "Rename %s to %s"
+		if opts.RenameLocalDir {
+			confirmMessage += " and the local directory name"
+		}
+		confirmMessage += "?"
+
 		var confirmed bool
 		if confirmed, err = opts.Prompter.Confirm(fmt.Sprintf(
-			"Rename %s to %s?", ghrepo.FullName(currRepo), newRepoName), false); err != nil {
+			confirmMessage, ghrepo.FullName(currRepo), newRepoName), false); err != nil {
 			return err
 		}
 		if !confirmed {
 			return nil
 		}
 	}
 
+	var repoDir string
+	if repoDir, err = opts.GitClient.ToplevelDir(context.Background()); err != nil {
+		return err
+	}
+	if opts.RenameLocalDir {
+		var exists bool
+		if exists, err = newRepoDirExists(repoDir, newRepoName); err != nil {
+			return err
+		}
+		if exists {
+			return fmt.Errorf("failed to rename local repository because the folder exists. Please choose another name")
+		}
+	}
+
 	apiClient := api.NewClientFromHTTP(httpClient)
 
 	newRepo, err := api.RenameRepo(apiClient, currRepo, newRepoName)
 	if err != nil {
 		return err
 	}
 
+	var renamedLocalDir bool
+	if opts.RenameLocalDir {
+		if err = renameRepoDir(repoDir, newRepoName); err == nil {
+			renamedLocalDir = true
+		} else {
+			fmt.Fprintf(opts.IO.ErrOut, "Failed to rename local directory: %v\n", err)
+		}
+	}
+
 	cs := opts.IO.ColorScheme()
 	if opts.IO.IsStdoutTTY() {
-		fmt.Fprintf(opts.IO.Out, "%s Renamed repository %s\n", cs.SuccessIcon(), ghrepo.FullName(newRepo))
+		if renamedLocalDir {
+			fmt.Fprintf(opts.IO.Out, "%s Renamed repository %s and local directory\n", cs.SuccessIcon(), ghrepo.FullName(newRepo))
+		} else {
+			fmt.Fprintf(opts.IO.Out, "%s Renamed repository %s\n", cs.SuccessIcon(), ghrepo.FullName(newRepo))
+		}
```

Comment:
There's some weird behavior in how the local dir interacts with the rename on mac, here:

```shell
# In the repo you are going to rename

$ pwd
<path>/original-repo-name

$ gh repo rename renamed-repo --rename-dir --yes
✓ Renamed repository <path>/renamed-repo and local directory
✓ Updated the "origin" remote

$ pwd
<path>/original-repo-name

$ ls <path>
renamed-repo
```

In other words, after the rename, the old directory still exists as long as we are in it. I think it may be helpful to do some navigating through the os so we end up in the target directory after the rename.

Reply:
My message is different from yours (I’m using Windows)😱. I have an additional error message about failing to rename:
```
$ gh repo rename renamed-repo --rename-dir --yes
Failed to rename local directory: rename C:/Users/user/Desktop/origin-repo C:\Users\user\Desktop\renamed-repo: The process cannot access the file because it is being used by another process.
✓ Renamed repository benebsiny/ltcd
✓ Updated the "origin" remote
```
---
> In other words, after the rename, the old directory still exists as long as we are in it. I think it may be helpful to do some navigating through the os so we end up in the target directory after the rename.

yeah, this is exactly the challenge when trying to resolve this issue